passes/pmgen/ice40_dsp.pmg

```diff
@@ -0,0 +1,160 @@
+state <SigBit> clock
+state <bool> clock_pol clock_vld
+state <SigSpec> sigA sigB sigY sigS
+state <Cell*> addAB muxAB
+
+match mul
+	select mul->type.in($mul)
+	select GetSize(mul->getPort(\A)) + GetSize(mul->getPort(\B)) > 10
+	select GetSize(mul->getPort(\Y)) > 10
+endmatch
+
+match ffA
+	select ffA->type.in($dff)
+	// select nusers(port(ffA, \Q)) == 2
```

Comment:
Is there a reason for this check being commented out?

Reply:
Yes. Even if the FF has multiple users we can still use it here because it's an input FF, so we can just in essence duplicate it. But we can't do that with the `D` input of `ffY`, so we need the check there.